gui/src/loader.rs

```diff
@@ -275,20 +277,32 @@ impl Loader {
     }
 
     pub fn subscription(&self) -> Subscription<Message> {
-        if let Some(Some(bitcoind_stdout)) =
-            self.internal_bitcoind.as_ref().map(|b| b.stdout.clone())
-        {
-            iced::subscription::unfold(0, bitcoind_stdout, move |stdout| async {
-                let msg = {
-                    let mut s = stdout.lock().await;
-                    let mut s = std::io::BufReader::new(s.deref_mut());
-                    let mut buffer = String::new();
-                    match s.read_line(&mut buffer) {
-                        Err(e) => Message::BitcoindLog(Some(e.to_string())),
-                        Ok(_) => Message::BitcoindLog(Some(buffer)),
+        if self.internal_bitcoind.is_some() {
+            let log_file = match read_end_bitcoind_log_file(&self.datadir_path, self.network) {
+                Ok(file) => file,
+                Err(e) => {
+                    tracing::warn!(
+                        "Cannot access to the end of the internal bitcoind log file: {}",
+                        e
+                    );
+                    return Subscription::none();
+                }
+            };
+
+            iced::subscription::unfold(0, log_file, move |mut log_file| async {
+                let mut buffer = [0u8; 1];
+                let mut last_line = String::new();
+                // Read bytes from the file and append them to the last_line string until a newline character is found
+                while log_file.read_exact(&mut buffer).is_ok() {
```

Comment:
Isn't this going to block until there is more content added to the file, since you seeked to the end?